src/query/plan/sql.py

```diff
@@ -33,12 +37,24 @@ def execute(self) -> pl.DataFrame:
                 f"""
                     Sending SQL query:
                         {sql_query}
-                """)
-            df = pl.read_database(sql_query, self.connection_uri)
+                """
+            )
+            # example of usage
+            if _SETTINGS.USE_CACHE:
+                df = _SETTINGS.CACHE.get_or_set(
+                    key=sql_query,
+                    call_back=pl.read_database,
+                    *sql_query,
+                    *self.connection_uri,
```

Comment:
It may a bit early for that but the API of this method seems really complicated, and leads to repeating with or without cache.
A simpler and maybe more readable way would simply to have a simple wrapper for the function like
```
df = use_cache(
  cache_type='sql_query',
  cache_key=[sql_query, self.connection_uri],
  method=lambda: pl.read_database(sql_query, self.connection_uri) 
)
```

This could even become a decorator `@use_cache` later on.

Reply:
really good idea !
my only concern is for the "cache_type" attr, what or why would we need that param ? 🤔 
Or it is supposed to indicate the appropriate Cache class we're going to use ?

Reply:
That could be both : 
- either used in keys to discriminate different type of caches inside the same storage
- or to use a different cache storage